### PR TITLE
feat: if scheduling is none, indexing must be none

### PR DESCRIPTION
**Pull request by jgpruitt**
we only support automatic index creation if scheduling is enabled. fail if indexing is not none but scheduling is none

### CODE REVIEW

**Review by alejandrodnm — APPROVED** [review]

**Review by JamesGuthrie — APPROVED** [review]

File: projects/extension/sql/idempotent/012-vectorizer-api.sql [hunk 1]
```diff
@@ -106,8 +106,15 @@ begin
     -- validate the scheduling config
     perform ai._validate_scheduling(scheduling);
 
+    -- validate the processing config
     perform ai._validate_processing(processing);
 
+    -- if scheduling is none then indexing must also be none
+    if scheduling operator(pg_catalog.->>) 'implementation' = 'none'
+    and indexing operator(pg_catalog.->>) 'implementation' != 'none' then
+        raise exception 'automatic indexing is not supported without scheduling. indexing must be none if scheduling is none';
```
Comment (alejandrodnm) [inline projects/extension/sql/idempotent/012-vectorizer-api.sql:115]: Sorry, should we be more clear and display `indexing=>ai.indexing_none()`?

Comment (jgpruitt) [reply projects/extension/sql/idempotent/012-vectorizer-api.sql]: updated. look okay to you @alejandrodnm ?

**Comment by jgpruitt** [comment]
rebased

File: projects/extension/tests/vectorizer/test_vectorizer.py [hunk 2]
```diff
@@ -1118,3 +1118,38 @@ def test_naming_collisions():
             );
             """)
             assert True
+
+
+def test_none_index_scheduling():
```
Comment (cevian) [inline projects/extension/tests/vectorizer/test_vectorizer.py:1123]: optional: add code to this test for success case. i.e. indexing_none() works

**Review by cevian — APPROVED** [review]

**Pull request merged**